src/index.js

```diff
@@ -9,60 +11,97 @@ import { persistRaw, persistSanitized, pushChanges } from './history/index.js';
 import sanitize from './sanitizer/index.js';
 import getServiceProviders, { getSanitizers } from './service_providers/index.js';
 import { DOCUMENTS_TYPES } from './documents_types.js';
-
-export async function updateServiceProviderDocument({ serviceProviderId, serviceProviderName, document }) {
-  const { documentType, url, contentSelector, sanitizationPipeline } = document;
-  const logPrefix = `[${serviceProviderName}-${DOCUMENTS_TYPES[documentType].name}]`;
-  console.log(`${logPrefix} Scrape '${url}'.`);
-  const content = await scrape(url);
-
-  const { sha: rawSha, filePath: rawFilePath} = await persistRaw(serviceProviderId, documentType, content);
-  if (rawSha) {
-    console.log(`${logPrefix} Save raw file to '${rawFilePath}'.`);
-    console.log(`${logPrefix} Commit raw file in '${rawSha}'.`);
-  } else {
-    console.log(`${logPrefix} No raw changes, didn't commit.`);
+export default class CGUs extends events.EventEmitter {
+  constructor() {
+    super();
+    this.initialized;
+    this.serviceProvidersManifests;
+    this.documentTypes = DOCUMENTS_TYPES;
   }
 
-  const sanitizers = getSanitizers(serviceProviderId);
-  const sanitizedContent = await sanitize(content, contentSelector, sanitizationPipeline, sanitizers);
+  async init() {
+    if (!this.initialized) {
+      this.serviceProvidersManifests = await getServiceProviders();
```

Comment:
The whole aim of a constructor is to initialize. Since we switch to a class syntax, I believe initialization should be started at instantiation, and a promise exposed on which to chain, so that we expose something equivalent to `new CGUs().isReady.then(…)`.

Reply:
To be discussed

Reply:
It seems to be a bad practice to execute async code in constructor (see [here](https://stackoverflow.com/questions/43431550/async-await-class-constructor) and [here](https://stackoverflow.com/questions/24398699/is-it-bad-practice-to-have-a-constructor-function-return-a-promise)) as the constructor should be a pure function.
So to expose something really simple to use, I suggest to do something like that:
````
export default class CGUs extends events.EventEmitter {
  constructor() {
    super();
    this.documentTypes = DOCUMENTS_TYPES;
  }

  async init() {
    if (!this.initialized) {
      this.serviceProvidersManifests = await getServiceProviders();
      this.initialized = Promise.resolve();
    }

    return this.initialized;
  }

  async updateTerms() {
    await this.init();

    console.log('Start scraping and saving terms of service…');
    
    […]
  }

  […]
}
```